tasks/download-atom-shell-task.coffee

```diff
@@ -98,71 +102,117 @@ module.exports = (grunt) ->
+        github.downloadAsset asset, (error, inputStream) ->
+          if error?
+            grunt.log.error "Cannot download atom-shell #{version}", error
+            return done false
+
+          # Save file to cache.
+          grunt.verbose.writeln "Downloading atom-shell #{version}."
+          downloadAndUnzip inputStream, path.join(versionDownloadDir, "atom-shell.zip"), (error) ->
             if error?
-              grunt.log.error "Cannot download atom-shell #{version}", error
+              grunt.log.error "Failed to download atom-shell #{version}", error
               return done false
 
-            # Save file to cache.
-            grunt.verbose.writeln "Downloading atom-shell #{version}."
-            saveAtomShellToCache inputStream, outputDir, downloadDir, version, (error) ->
-              if error?
-                grunt.log.error "Failed to download atom-shell #{version}", error
-                return done false
+            grunt.verbose.writeln "Installing atom-shell #{version}."
+            copyDirectory(versionDownloadDir, outputDir)
+            rebuildNativeModules apm, currentAtomShellVersion, version, rebuild, done
+        return
+
+      grunt.log.error "Cannot find #{filename} in atom-shell #{version} release"
+      done false
+
+  grunt.registerTask "#{TaskName}-chromedriver", 'Download the chromedriver binary distributed with atom-shell',  ->
+    @requiresConfig "#{TaskName}.version", "#{TaskName}.outputDir"
+    {version, outputDir, downloadDir, token} = grunt.config(TaskName)
+    version = "v#{version}"
+    downloadDir ?= path.join os.tmpdir(), 'downloaded-atom-shell'
+    chromedriverPath = path.join(outputDir, "chromedriver")
+
+    done = @async()
+
+    # Chromedriver is only distributed with the first patch release for any
+    # given major and minor version of atom-shell.
+    versionWithChromedriver = version.split(".").slice(0, 2).join(".") + ".0"
+    downloadPath = path.join(downloadDir, "#{versionWithChromedriver}-chromedriver")
+
+    # Do nothing if the desired version of atom-shell is already installed with
+    # a chromedriver executable.
+    currentAtomShellVersion = getAtomShellVersion(outputDir)
+    return done() if currentAtomShellVersion is version and grunt.file.isDir(chromedriverPath)
+
+    # Use a cached download of chromedriver if one exists.
+    if grunt.file.isDir(downloadPath)
+      grunt.verbose.writeln("Installing cached chromedriver #{versionWithChromedriver}.")
+      copyDirectory(downloadPath, chromedriverPath)
+      return done()
+
+    # Request the assets.
+    github = new GitHub({repo: 'atom/atom-shell', token})
+    github.getReleases tag_name: versionWithChromedriver, (error, releases) ->
+      unless releases?.length > 0
+        grunt.log.error "Cannot find atom-shell #{versionWithChromedriver} from GitHub", error
+        return done false
+
+      assetNameRegex = new RegExp("chromedriver-.*-#{process.platform}-#{getArch()}")
```

Comment:
Super minor, but the CoffeeScript interpolated regexes using `///` makes things a litter terser.

```coffee
assetNameRegex = ///chromedriver-.*-#{process.platform}-#{getArch()}"///
```